### PR TITLE
Abstract block info

### PR DESCRIPTION
**Pull request by syntrust**
When mining on L2 sometimes run into "StorageContract: minedTs too small". 

This could be an edge case when a blob in a new shard happens to be mined immediately after the shard is opened by `pubBlobs`.

The test can be done by running the following cmd with code https://github.com/ethstorage/es-node/pull/187
```
env ES_NODE_SIGNER_PRIVATE_KEY=<pk> ./integration_tests/run_tests.sh
```

The reason is that miningInfo is timestamped by L2 chain which has block time of 2 seconds while mine method computes minedTs with L1 blocks of 12 seconds block time.

This PR abstracts blockNumber() and blockTs() as well as getRandao, etc. for StorageContract and its L2 extender so that they can use different values and implementations.

### CODE REVIEW

File: contracts/EthStorageContractL2.sol [hunk 1]
```diff
@@ -35,55 +35,23 @@ contract EthStorageContractL2 is EthStorageContract2 {
...
+    }
+
+    /// @notice Get the current block timestamp
+    function blockTs() internal view override returns (uint256) {
+        return L1_BLOCK.timestamp();
+    }
+
     /// @notice Get the randao value from the L1 blockhash.
-    function _getRandao(uint256 _l1BlockNumber, bytes calldata _headerRlpBytes) internal view returns (bytes32) {
+    function getRandao(
```
Comment (qzhodl) [inline contracts/EthStorageContractL2.sol:49]: Since we are using forge and hardhat at the same time, let's change the formatter to forge together (vscode => settings => solidity:formatter => forge)

File: contracts/test/TestEthStorageContract.sol [hunk 2]
```diff
@@ -145,12 +145,9 @@ contract TestEthStorageContract is EthStorageContract {
         bytes[] calldata inclusiveProofs,
         bytes[] calldata decodeProof
     ) internal {
-        // Obtain the blockhash of the block number of recent blocks
-        require(block.number - blockNumber <= 64, "block number too old");
-        // To avoid stack too deep, we resue the hash0 instead of using randao
-        bytes32 hash0 = RandaoLib.verifyHistoricalRandao(blockNumber, randaoProof);
-        // Estimate block timestamp
-        uint256 mineTs = block.timestamp - (block.number - blockNumber) * 12;
+        require(super.blockNumber() - blockNumber <= MAX_L1_MINING_DRIFT, "block number too old");
```
Comment (qzhodl) [inline contracts/test/TestEthStorageContract.sol:148]: Why use super here?

File: contracts/StorageContract.sol [hunk 3]
```diff
@@ -396,6 +387,28 @@ abstract contract StorageContract is DecentralizedKV {
...
+
+    /// @notice Get the randao value by block number.
+    function _getRandao(uint256 _blockNum, bytes calldata _headerRlpBytes) internal view virtual returns (bytes32) {
+        bytes32 bh = blockhash(_blockNum);
+        require(bh != bytes32(0), "StorageContract: failed to obtain blockhash");
+        return RandaoLib.verifyHeaderAndGetRandao(bh, _headerRlpBytes);
+    }
+
+    /// @notice Get the mined timestamp
+    function getMinedTs(uint256 _blockNum) internal view returns (uint256) {
```
Comment (qzhodl) [inline contracts/StorageContract.sol:408]: We should also follow the internal function naming convention for this one

**Review by qzhodl — APPROVED** [review]

**Review by iteyelmp — APPROVED** [review]
